src/main/java/mate/academy/spring/controller/UserController.java

```
@@ -0,0 +1,28 @@
+package mate.academy.spring.controller;
+
+import mate.academy.spring.mapper.DtoResponseMapper;
+import mate.academy.spring.model.User;
+import mate.academy.spring.model.dto.response.UserResponseDto;
+import mate.academy.spring.service.UserService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+    private DtoResponseMapper<UserResponseDto, User> userResponseMapper;
+    private UserService userService;
+
+    public UserController(DtoResponseMapper<UserResponseDto, User> userResponseMapper,
+                          UserService userService) {
+        this.userResponseMapper = userResponseMapper;
+        this.userService = userService;
+    }
+
+    @GetMapping("/by-email")
+    public UserResponseDto getByEmail(@RequestParam String email) {
+        return userResponseMapper.toDto(userService.findByEmail(email).orElseThrow());
```

Let's throw an exception and add the informative message about the error here.